plugins/ltac2/tac2core.ml

```diff
@@ -1043,10 +1043,43 @@ let () =
   define "time" (option string @-> closure @-> tac valexpr) @@ fun s f ->
   Proofview.tclTIME s (thaw f)
 
+let () =
+  define "timeout" (int @-> closure @-> tac valexpr) @@ fun i f ->
+    Proofview.tclTIMEOUT i (Tac2ffi.apply f [Tac2ffi.of_unit ()])
+
+let () =
+  define "timeoutf" (float @-> closure @-> tac valexpr) @@ fun f64 f ->
+    Proofview.tclTIMEOUTF (Float64.to_float f64) (Tac2ffi.apply f [Tac2ffi.of_unit ()])
+
 let () =
   define "check_interrupt" (unit @-> tac unit) @@ fun _ ->
   Proofview.tclCHECKINTERRUPT
 
+let () =
+  define "throw_bt" (exn @-> exninfo @-> tac valexpr) @@ fun (e,_) info ->
+    Proofview.tclLIFT (Proofview.NonLogical.raise (e, info))
+
+let () =
+  define "zero_bt" (exn @-> exninfo @-> tac valexpr) @@ fun (e,info0) info ->
+    let bt = Exninfo.get info Tac2bt.backtrace in
+    let fn info = Exninfo.add info Tac2bt.backtrace in
+    Proofview.tclZERO ~info:(Option.fold_left fn info0 bt) e
```

Comment:
the option combinators are hard to read IMO, please write this as a regular match